lambda/action_network.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import os
+from typing import Dict
+
+import requests
+from dateutil import parser
+
+log = logging.getLogger(__name__)
+
+
+def get_object_or_empty_dict(key: str, event: Dict) -> Dict:
+    location = event.get(key)
+    if not location:
+        location = {}
+    return location
+
+
+def make_location(event: Dict) -> str:
+    location = get_object_or_empty_dict('location', event)
+    if location.get('country', 'US') != 'US':
+        return '{city}, {country}'.format(
+            city=location.get('locality'), country=location.get('country'))
+    return location.get('postal_code', '')
+
+
+def get_event_name(event: Dict) -> str:
+    name = event.get('name', None)
+    if not name:
+        name = event.get('title', '')
+    return name
+
+
+def get_organizer(event: Dict) -> Dict:
+    return get_object_or_empty_dict('osdi:organizer',
+                                    get_object_or_empty_dict(
+                                        '_embedded', event))
+
+
+def get_contact_name(event: Dict) -> Dict:
+    organizer = get_organizer(event)
+    given_name = organizer.get('given_name')
+    family_name = organizer.get('family_name')
+    return ' '.join([given_name, family_name])
+
+
+def get_email_address_from_organizer(organizer: Dict) -> str:
+    email_addresses = organizer.get('email_addresses', {})
+    next_best_email_address = ''
+    for this_email_address in email_addresses:
+        address = this_email_address.get('address')
+        if this_email_address.get('primary'):
+            return address
+        elif not next_best_email_address:
+            next_best_email_address = address
+
+    return next_best_email_address
+
+
+def get_email_address(event: Dict) -> str:
+    return get_email_address_from_organizer(get_organizer(event))
+
+
+def get_events_from_events_campaign(return_events=None,
+                                    page=None) -> Dict[str, Dict]:
+    if not return_events:
```

Comment:
use the defaults in the function definition (`return_events={}, page=1`); no need to set defaults and then re-set them to different values

Reply:
Using {} as a default argument is a bad idea.  The next time the function is called, it will still have items populated from the current invocation.  Agree about page.